File: parcel-locker-cpp/library/src/model/DepositBox.cpp
//
// Created by student on 31.05.2021.
//

#include <model/DepositBox.h>


std::string DepositBox::getId() {
    return id;
}

bool DepositBox::getStatus() {
    return status;
}

bool DepositBox::Access(const std::string &code, const std::string &telNumber) {
    if(code == this->accessCode && telNumber == this->telNumber && !code.empty() && !telNumber.empty()) {
        status = false;
        this->accessCode = "";
        this->telNumber = "";
        return true;
    }
    return false;
}


DepositBox::DepositBox(const std::string &id): id(id) {
    status = false;
    telNumber = "";
    accessCode = "";
}


void DepositBox::setAccessCode(const std::string &code) {
    accessCode = code;
}

void DepositBox::setTelNumber(const std::string &numer) {
    telNumber = numer;
}

void DepositBox::setStatus(bool dostepnosc) {
    status = dostepnosc;
}

std::string DepositBox::getTelNumber() {
    return telNumber;
}

std::string DepositBox::getAccessCode() {
    return accessCode;
}

void DepositBox::setDeliveryId(const boost::uuids::uuid &deliveryId) {
    DepositBox::deliveryId = deliveryId;
}

const boost::uuids::uuid &DepositBox::getDeliveryId() const {
    return deliveryId;
}

File: parcel-locker/src/main/java/pl/pas/parcellocker/model/locker/DepositBox.java
package pl.pas.parcellocker.model.locker;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.EqualsAndHashCode;
import pl.pas.parcellocker.model.EntityModel;
import pl.pas.parcellocker.model.delivery.Delivery;

import java.util.UUID;

@Entity
@EqualsAndHashCode
public class DepositBox extends EntityModel {

    @ManyToOne
    @JoinColumn(name = "delivery_id")
    private Delivery delivery;
    private boolean isEmpty;
    private String accessCode;
    private String telNumber;

    public DepositBox() {
        isEmpty = true;
        telNumber = "";
        accessCode = "";
    }

    public boolean canAccess(String code, String telNumber) {
        return code.equals(this.accessCode)
            && telNumber.equals(this.telNumber)
            && !code.isEmpty()
            && !telNumber.isEmpty();
    }

    public void putIn(Delivery delivery, String telNumber, String accessCode) {
        this.accessCode = accessCode;
        this.isEmpty = false;
        this.telNumber = telNumber;
        this.delivery = delivery;
    }

    public void clean() {
        isEmpty = true;
        this.accessCode = "";
        this.telNumber = "";
        this.delivery = null;
    }

    public Delivery getDelivery() {
        return delivery;
    }

    public boolean isEmpty() {
        return isEmpty;
    }

    public String getAccessCode() {
        return accessCode;
    }

    public String getTelNumber() {
        return telNumber;
    }

    public void setAccessCode(String accessCode) {
        this.accessCode = accessCode;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public void setTelNumber(String telNumber) {
        this.telNumber = telNumber;
    }
}
